numba/unicode.py

```diff
@@ -1361,6 +1361,25 @@ def impl(a):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L12017-L12045    # noqa: E501
+@overload_method(types.UnicodeType, 'isdecimal')
+def unicode_isdecimal(data):
+    """Implements UnicodeType.isdecimal()"""
+
+    def impl(data):
+        length = len(data)
+        if length == 0:
+            return False
+
```

Comment:
The CPython impl has a single char shortcut https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L12031-L12034 is there merit in implementing this?